src/v/raft/consensus.cc

```diff
@@ -358,6 +358,12 @@ consensus::success_reply consensus::update_follower_index(
           _group));
     }
 
+    // check preconditions for processing the reply
+    if (unlikely(!is_elected_leader())) {
+        vlog(_ctxlog.debug, "ignoring append entries reply, not leader");
+        return success_reply::no;
+    }
+
     update_node_reply_timestamp(node);
 
     if (
```

Comment:
BTW why do we need the `reply.last_dirty_log_index < _log.offsets().dirty_offset` condition? Is it because we still need to process a reply of a leader in some later term? Maybe we need to move the term check before this conditional too?

Reply:
from the slack discussion: this condition means that processing this response won't trigger recovery, so processing it will result in progress even if it is reordered